modules/promotion/src/Element/CouponRedemptionForm.php

```diff
@@ -224,12 +289,98 @@ public static function validateForm(array &$element, FormStateInterface $form_st
       $form_state->setErrorByName($code_path, t('Coupon is invalid'));
       return;
     }
-    if (!$coupon->getPromotion()->applies($order)) {
-      $form_state->setErrorByName($code_path, t('Coupon is invalid'));
-      return;
-    }
 
     $form_state->setValueForElement($element, $coupon);
   }
 
+  /**
+   * Adjustments table builder.
+   *
+   * @param array $element
+   *   The element.
+   * @param \Drupal\commerce_order\Entity\OrderInterface $order
+   *   The order.
+   * @param bool $display_actions
+   *   TRUE if actions displayed.
+   *
+   * @return array
+   *   The element array.
+   */
+  public static function buildAdjustmentsTable(array $element, OrderInterface $order, $display_actions = TRUE) {
```

Comment:
@mglaman It might be useful if there were a separate render element for listing coupons, so that it could easily be rendered outside of the coupon redemption form as well.

As an example use case, I'm trying to get almost the exact same table (but without action buttons, giving one use case for that parameter) into the order receipt template so that it can be included in order emails.